src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs

```diff
@@ -123,11 +124,12 @@ private static TMetrics GetAverageMetrics(IEnumerable<TMetrics> metrics, TMetric
                     logLoss: GetAverageOfNonNaNScores(newMetrics.Select(x => x.LogLoss)),
                     logLossReduction: GetAverageOfNonNaNScores(newMetrics.Select(x => x.LogLossReduction)),
                     topKPredictionCount: newMetrics.ElementAt(0).TopKPredictionCount,
-                    topKAccuracy: GetAverageOfNonNaNScores(newMetrics.Select(x => x.TopKAccuracy)),
-                    // Return PerClassLogLoss and ConfusionMatrix from the fold closest to average score
+                    //need to "transpose"/rotate this array of arrays so we can average across all top 0, top 1, top 2, etc
+                    topKAccuracies: newMetrics.SelectMany(nm => nm.TopKAccuracyForAllK.Select((double tk, int i) => (i, tk))).ToLookup(itk => itk.i, itk => itk.tk).Select(tk => GetAverageOfNonNaNScores(tk)).ToArray(),
```

Comment:
I believe this is actually doing the same thing as `GetAverageOfNonNaNScoresInNestedEnumerable`?
Or maybe I'm missing something?

Here's the definition of that method, and then the 2 places where it's called:

https://github.com/dotnet/machinelearning/blob/c255ac7c45af7a19a6e5ab1e26fdbe44874f241a/src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs#L161-L170

https://github.com/dotnet/machinelearning/blob/c255ac7c45af7a19a6e5ab1e26fdbe44874f241a/src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs#L152-L156

Reply:
//cc: @justinormont , since this is modifying AutoML can you please approve these changes?

BTW, I think there's a bug on `GetAverageOfNonNaNScoresInNestedEnumerable`, I don't see why the assertion is needed:
https://github.com/dotnet/machinelearning/blob/c255ac7c45af7a19a6e5ab1e26fdbe44874f241a/src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs#L166
it seems also wrong, if we have, say 3 DCG arrays coming from 3 different RankingMetrics objects, and each array is 10 elements long, then the assertion will fail when i = 10. @mstfbl , I think you wrote this method, am I correct about this? thanks.

Reply:
Agree, that assert is wrong

Reply:
Hey all, that is true in that the assert statement would fail when `i=4` with say an array of 10 elements long. Thank you for pointing that out.

Reply:
@jasallen can you please use this method and remove the assertion? thanks! 😄